src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -871,6 +871,16 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
         }
 
         [Fact]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_R()
+        {
+            DateTimeOffset expected = DateTimeOffset.MaxValue;
+            string expectedString = expected.ToString("r");
+
+            DateTimeOffset result = DateTimeOffset.ParseExact(expectedString, "r", null, DateTimeStyles.None);
+            Assert.Equal(expectedString, result.ToString("r"));
+        }
```

Comment:
This seems to be testing that we can successfully roundtrip via ToString and ParseExact.  Do we also have tests that verify we get the expected string from ToString?  Do we have tests that verify this for a variety of inputs, i.e. various years/months/days/hours/minutes/seconds/offsets/etc.?  If not, you might consider making this a theory that takes both the DateTimeOffset and the expected string, and then having the test do what it's doing now but also compare the generated ToString to the expected one provided. (Same comment/question for the DateTime test later.)